semantic/parse.go

```diff
@@ -0,0 +1,63 @@
+// Package semantic provides version parsing and comparison for various ecosystems,
+// matching the native versioning rules of each ecosystem.
+package semantic
+
+import (
+	"errors"
+	"fmt"
+)
+
+var ErrUnsupportedEcosystem = errors.New("unsupported ecosystem")
+
+// MustParse is like Parse but panics if the ecosystem is not supported.
+func MustParse(str string, ecosystem string) Version {
+	v, err := Parse(str, ecosystem)
+
+	if err != nil {
+		panic(err)
+	}
+
+	return v
+}
+
+// Parse attempts to parse the given string as a version for the specified ecosystem,
+// returning an ErrUnsupportedEcosystem error if the ecosystem is not supported.
+func Parse(str string, ecosystem string) (Version, error) {
+	//nolint:exhaustive // Using strings to specify ecosystem instead of lockfile types
+	switch ecosystem {
+	case "Alpine":
```

Comment:
yeah this is a point to be discussed w/ @another-rex and co, since these are currently OSV ecosystems - `string` was used  over the `Ecosystem` to avoid a cycle import as that particular type has had a mess (and duplicate) history which we've been working to improve

Reply:
Got it. Leaving to your and @another-rex to hash out. Feel free to resolve since this isn't particularly important in this context :)